tests/unit/multieditor/test_actions.py

```diff
@@ -0,0 +1,281 @@
+    }
+    expected_map = {
+      'key_a': [{'key_c': ['val5', 'val4']},
+                {'key_c': ['val4']},
+                {'key_c': ['val3']}],
+      'key_b': {'key_c': {'key_d': 'pong'}}
+    }
+    actions_tuple = Action(values_to_check=["val6", "val1"], keys=['key_a', 'key_c'], where_keys=[],
+                           selected_action="Deletion",
+                           where_value="", regex=False, value="")
+    assert actions.run_action({}, record, actions_tuple) == expected_map
+
+
+def test_deletion_empty_rec():
+    record = {
+        'key1': {
+            'key2': {
+                'key3': 'val'
+            }
+        }
+    }
+    expected_map = {}
+    actions_tuple = Action(values_to_check=[], keys=['key1', 'key2', 'key3'], where_keys=[], selected_action="Deletion",
+                           where_value="", regex=False, value="")
+
+    assert actions.run_action({}, record, actions_tuple) == expected_map
+
+
+def test_field_not_existing():
+    """should test sub_record creation for missing object"""
+    record = {'abstracts': [{'not_source': 'success'}]}
+    expected_map = {'abstracts': [{'not_source': 'success'}]}
+    actions_tuple = Action(values_to_check=[], keys=['abstracts', 'source'], where_keys=[], selected_action="Update",
+                           where_value="", regex=False, value="success")
+    assert actions.run_action({}, record, actions_tuple) == expected_map
+
+
+def test_record_creation():
+    """should test sub_record creation for missing object"""
+    key = ['abstracts', 'source']
+    value = 'success'
+    target_object = {'abstracts': [{'source': 'success'}]}
+    assert actions.create_schema_record(schema_1, key, value) == target_object
+
+
+def test_record_creation_2():
+    """should test sub_record creation for missing object"""
+    schema_2 = {
+        "properties": {
+            "source": {
+                "$schema": "http://json-schema.org/schema#",
+                "description": "Source of the information in this field. As several\
+                            records can be merged,\nthis information\
+                            allows us to remember where every bit of\
+                            metadata came\nfrom and make decisions based\
+                            on it.\n\n:MARC: Often not present.",
+                "type": "string"
+            }},
+        "type": "object",
+    }
+    key = ['source']
+    value = 'success'
+    target_object = {'source': 'success'}
+    assert actions.create_schema_record(schema_2, key, value) == target_object
+
+
+def test_record_creation_3():
+    """should test sub_record handling for missing object"""
+    record_1 = {
+        "abstracts": [
+            {
+                "value": "A dataset corresponding to $2.8~\\mathrm{fb}^{-1}$"
+            }
+        ],
+    }
+    target_object = {
+        "abstracts": [
+          {
+            "value": "A dataset corresponding to $2.8~\\mathrm{fb}^{-1}$"
```

Comment:
Since you are showing that values are the same I would put a simpler example. Otherwise it takes a moment to make sure that formula is the same ;)